chia/util/streamable.py

```diff
@@ -134,54 +120,38 @@ def dataclass_from_dict(klass: Type[Any], d: Any) -> Any:
         return klass(d)
 
 
-@overload
-def recurse_jsonify(d: Union[List[Any], Tuple[Any, ...]]) -> List[Any]:
-    ...
-
-
-@overload
-def recurse_jsonify(d: Dict[str, Any]) -> Dict[str, Any]:
-    ...
-
-
-def recurse_jsonify(d: Union[List[Any], Tuple[Any, ...], Dict[str, Any]]) -> Union[List[Any], Dict[str, Any]]:
+def recurse_jsonify(d: Any) -> Any:
```

Comment:
Could we maybe keep the hinting and just add a `Any` -> `Dict[str, Any]` to the end?  I think that would just be a catch all if the existing overloads don't match.

Reply:
I might need some help with the hinting. The return type isn't necessarily a `Dict[]`, it's something we might call "JSONCompatible" i.e. one of `Dict[str, JSONCompatible]`, `List[JSONCompatible]`, 'int', 'str'

Reply:
at least the simple type definition:
```
JsonCompat = Union[Dict[str, "JsonCompat"], List["JsonCompat"], str, int, None]
```
makes mypy fail with:
``` 
error: Cannot resolve name "JsonCompat" (possible cyclic definition)  [misc]
```